torch/lib/TH/generic/THTensorMath.c

```diff
@@ -2829,26 +2829,84 @@ TENSOR_IMPLEMENT_LOGICAL(ge,>=)
 TENSOR_IMPLEMENT_LOGICAL(eq,==)
 TENSOR_IMPLEMENT_LOGICAL(ne,!=)
 
+#define VECTOR_IMPLEMENT_FUNCTION(NAME, CFUNC)  \
+  void THVector_(NAME)(real *y, const real *x, const ptrdiff_t n) \
+  { \
+    ptrdiff_t i = 0;  \
+    for(; i<n-4; i+=4)  \
+    { \
+      y[i] = CFUNC(x[i]); \
+      y[i+1] = CFUNC(x[i+1]); \
+      y[i+2] = CFUNC(x[i+2]); \
+      y[i+3] = CFUNC(x[i+3]); \
+    } \
+    for(; i < n; i++) \
+      y[i] = CFUNC(x[i]); \
+  } \
+
+#define VECTOR_IMPLEMENT_FUNCTION_VALUE(NAME, CFUNC)  \
```

Comment:
I think there's no need to unroll these loops yourself. The compiler should be able to do it.